pkg/process/tagger/tagger.go

```diff
@@ -43,19 +48,182 @@ func findCandidateTags(appVersion *runtime.AuroraVersion, tagOverwrite bool, out
+				appVersion, repositoryTags)
+		}
+		filteredTags, err := filterTagsFromRepository(appVersion, candidateTags, repositoryTags)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Error in FilterVersionTags, app_version=%s, "+
+				"candidateTags=%v, repositoryTags=%v", appVersion, candidateTags, repositoryTags)
+		}
+		return filteredTags, nil
+	} else {
+		versions := make([]string, 0, 10)
+		logrus.Debug("Is not semantic version. Append only complete version and given version")
+		versions = append(versions, string(appVersion.GetCompleteVersion()))
+		if appVersion.Snapshot {
+			versions = append(versions, string(appVersion.GetGivenVersion()))
+		}
+		return versions, nil
+	}
+}
+
+func filterTagsFromRepository(version *runtime.AuroraVersion, candidateTags []string,
+	repositoryTags []string) ([]string, error) {
+
+	var excludeMinor, excludeMajor, excludeLatest = true, true, true
+
+	minorTagName, err := getMinor(string(version.GetAppVersion()), true)
+	if err != nil {
+		return nil, err
+	}
+	excludeMinor, err = tagCompare("> "+string(version.GetAppVersion())+", < "+minorTagName, repositoryTags)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Debugf("Minor tag name: %s. Exclude: %t", minorTagName, excludeMinor)
+
+	majorTagName, err := getMajor(string(version.GetAppVersion()), true)
+	if err != nil {
+		return nil, err
+	}
+
+	excludeMajor, err = tagCompare("> "+string(version.GetAppVersion())+", < "+majorTagName, repositoryTags)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Debugf("Major tag name: %s. Exclude: %t", majorTagName, excludeMajor)
+
+	excludeLatest, err = tagCompare("> "+string(version.GetAppVersion()), repositoryTags)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Debugf("Exclude latest: %t", excludeLatest)
+
+	versions := make([]string, 0, 10)
+	sort.StringSlice(candidateTags).Sort()
+	for _, tag := range candidateTags {
+		if strings.EqualFold(strings.TrimSpace(tag), "latest") {
+			if !excludeLatest {
+				versions = append(versions, tag)
+			}
+		} else if isMinor(tag) {
+			if !excludeMinor {
+				versions = append(versions, tag)
+			}
+		} else if isMajor(tag) {
+			if !excludeMajor {
+				versions = append(versions, tag)
+			}
+		} else {
+			versions = append(versions, tag)
+		}
+	}
+	return versions, nil
+}
+
+func tagCompare(versionConstraint string, tags []string) (bool, error) {
+	c, err := extVersion.NewConstraint(versionConstraint)
+
+	if err != nil {
+		return false, errors.Wrapf(err, "Could not create version constraint %s", versionConstraint)
+	}
+	for _, tag := range tags {
+		v, err := extVersion.NewVersion(tag)
+
+		if err != nil {
+			// We won't fail on random tags in the reposiority
+			continue
+		}
+
+		if c.Check(v) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func getSemtanticVersionTags(version *runtime.AuroraVersion, extraTags config.PushExtraTags) ([]string, error) {
+	versions := make([]string, 0, 10)
```

Comment:
begrenser dette antall tags til 10?

Reply:
Nei, det setter slicen til å ha en underliggende kapasitet på 10, og lengde på 0.

Det gjør at man kan legge til 10 elementer uten at man trenger å allokere minne på nytt. Når den vokser over, vil det allokeres et nytt med append.

https://blog.golang.org/go-slices-usage-and-internals (søk etter append)